packages/material-ui/src/ClickAwayListener/ClickAwayListener.js

```diff
@@ -111,14 +122,52 @@ const ClickAwayListener = React.forwardRef(function ClickAwayListener(props, ref
     return undefined;
   }, [handleClickAway, mouseEvent]);
 
-  return <React.Fragment>{React.cloneElement(children, { ref: handleRef })}</React.Fragment>;
+  // Keep track of mouse/touch events that bubbled up through the portal.
+  const handleSyntheticMouse = (event) => {
```

Comment:
An edge case like `event.stopPropagation()` (synthetic) would cause this handler to be skipped, but the `document` one will still run because it's native, causing the click away to be fired still, but perhaps it's not worrying about..

Reply:
@oliviertassinari do I just resolve this one?

Reply:
Given that it only impact portaled element, the impact is limited. Also, developers still have the workaround to stop propagation on both synthetic and DOM events.